python/ray/dag/dag_node_operation.py

```diff
@@ -368,3 +505,72 @@ def _generate_actor_to_execution_schedule(
     for _, candidates in actor_to_candidates.items():
         assert len(candidates) == 0
     return actor_to_execution_schedule
+
+
+def _optimize_execution_schedule(
+    actor_to_execution_schedule: Dict[
+        "ray.actor.ActorHandle", List[_DAGOperationGraphNode]
+    ],
+    overlap_gpu_communication: bool,
+) -> Dict["ray.actor.ActorHandle", List[_DAGOperationGraphNode]]:
+    """
+    Optimize the execution schedule by overlapping computation and communication.
```

Comment:
based on this comment, why do we need `overlap_gpu_communication ` as a part of the argument? Should we just not call this function if overlap_gpu_communication is False?

Reply:
refactored.